src/Util/YamlSourceManipulator.php

```diff
@@ -449,8 +457,15 @@ private function changeValueInYaml($value)
 
         $endValuePosition = $this->findEndPositionOfValue($originalVal);
 
-        $newYamlValue = $this->convertToYaml($value);
-        if (!\is_array($originalVal) && \is_array($value)) {
+        $isMultilineValue = null !== $this->findPositionOfMultilineCharInLine($this->currentPosition);
+        $originalValEndsWithNewLine = $isMultilineValue ? rtrim($originalVal, "\n") !== $originalVal : false;
+
+        // In case of multiline, $value is converted as plain string like "Foo\nBar"
+        // We need to keep it "as is"
+        $newYamlValue = $isMultilineValue ? rtrim($value, "\n") : $this->convertToYaml($value);
```

Comment:
Why do we need to `rtrim()` here? I can see that if I remove it the tests fail, but I can't figure out why it's needed.

Reply:
One thing I noticed is, in this test case [multiline_string_before_property.test](https://github.com/symfony/maker-bundle/pull/610/files#diff-c137fbbe9d94d4a5f4bc29116ea1273e), `$value` ends with an `\n`

![image](https://user-images.githubusercontent.com/17051512/82738762-8730ae80-9d3a-11ea-9c13-3200433facf8.png)

This will cause `indentMultilineYamlArray` will add an extra line that's not wanted.

Reply:
Ah, I understand. I think the implementation is what's confusing me. All multiline arrays must end in a line break, by their very nature. If anything, in this spot, (just thinking about the logical flow of the code) I would expect you to be guaranteeing that the `$newYamlValue` *does* end in `\n`, instead of stripping it. After all, if we made sure that `$newYamlValue` ended in 1 line break... then that should work perfectly with YAML, which *also* wants the line break :). I'm wondering if we're trimming and adding `\n` in so many places that we're doing and undoing extra work.

There *is* the edge-case of not needing the "\n" if the multiline string is at the end of the file. That should be handled (somewhere) with an `if` statement specifically for it. Something like:

```php
// maybe down before the `$newContents = substr(...)` line
if (false === $this->findNextLineBreak($newPosition) && $isMultilineValue) {
    $newYamlValue = trim($newYamlValue, "\n");
}
```

Reply:
I won't lie to you, I also think there is a lot of trimming to handle multiline string...

Actually, it seems that re-adding original \n (after `$newYamlValue = "\n".$this->indentMultilineYamlArray($newYamlValue);`) breaks test case because, as mentionned above, YamlDumper will add an extra line

![image](https://user-images.githubusercontent.com/17051512/82753174-fa323780-9dc3-11ea-8fe9-452e8ed10619.png)

Maybe this bug is related to Yaml component and not `YamlSourceManipulator`